src/core_modules/capture-core/components/DataEntry/withAskToCreateNew.js

```diff
@@ -0,0 +1,99 @@
+// @flow
+import * as React from 'react';
+import i18n from '@dhis2/d2-i18n';
+import { Modal, ModalTitle, ModalContent, ModalActions, ButtonStrip, Button } from '@dhis2/ui';
+import { type RenderFoundation } from '../../metaData';
+import { addEventSaveTypes } from '../WidgetEnrollmentEventNew/DataEntry/addEventSaveTypes';
+
+type Props = {
+    onCancelCreateNew: (itemId: string) => void,
+    onConfirmCreateNew: (itemId: string) => void,
+    onSave: (eventId: string, dataEntryId: string, formFoundation: RenderFoundation, saveType?: ?string) => void,
+    allowGenerateNextVisit?: ?boolean,
+    isCompleted?: ?boolean,
+    itemId: string
+};
+
+type State = {
+    isOpen: boolean,
+}
+
+const askToCreateNewComponent = (InnerComponent: React.ComponentType<any>) =>
+    class AskToCreateNewHOC extends React.Component<Props, State> {
+        handleOnSave: () => void;
+        innerInstance: any;
+        constructor(props: Props) {
+            super(props);
+            this.state = {
+                isOpen: false,
+            };
+            this.handleOnSave = this.handleOnSave.bind(this);
+        }
+
+        getWrappedInstance() {
+            return this.innerInstance;
+        }
+
+        handleOnSave(eventId: string, dataEntryId: string, formFoundation: RenderFoundation, saveType?: string) {
+            if (this.props.allowGenerateNextVisit &&
+                (this.props.isCompleted || saveType === addEventSaveTypes.COMPLETE)) {
+                this.setState({ isOpen: true });
```

Comment:
Hey,
Should the modal be shown if there are no available stages where a new event can be added? This edge case occurs when the following conditions are met:
- all stages have `repeatable` false 
- all stages already have an event
- the stage where the event is completed has `allowGenerateNextVisit` true.  

https://user-images.githubusercontent.com/6817073/199425964-274a65a1-4c17-4ba0-8367-0ac16afd2d4f.mov


Does it make sense to check that there is at least one available stage? 
Thanks!